### PR TITLE
Update Carbon Calculator data in environmental impact guide

### PR DESCRIPTION
**Pull request by delucis**
#### Description

This PR updates the data in the comparison table in our environmental impact guide.

- Updates CO₂ numbers (last update was back in May 2023)
- Adds a “Rating” column to show letter grades from Website Carbon Calculator’s new [rating system](https://www.websitecarbon.com/introducing-the-website-carbon-rating-system/)
- Adds Mintlify to the table

Preview: https://deploy-preview-2136--astro-starlight.netlify.app/environmental-impact/#comparisons

### CODE REVIEW

**Review by sarah11918 — APPROVED** [review]
Looks great! A nice visual addition to give some context to the numbers.

**Review by TheOtterlord — APPROVED** [review]
Looks perfect!

File: docs/src/content/docs/environmental-impact.md [hunk 1]
```diff
@@ -89,20 +89,21 @@ Choosing a host that uses renewable energy will mean lower carbon emissions for
...
+| [VitePress][vp-carbon]      | 0.07g              |   A+   |
+| [Docus][dc-carbon]          | 0.09g              |   A+   |
+| [docsify][dy-carbon]        | 0.10g              |   A    |
+| [Nextra][nx-carbon]         | 0.11g              |   A    |
+| [MkDocs][mk-carbon]         | 0.19g              |   B    |
+| [Docusaurus][ds-carbon]     | 0.21g              |   B    |
+| [GitBook][gb-carbon]        | 0.43g              |   C    |
+| [Mintlify][mt-carbon]       | 1.22g              |   F    |
+
+<small>Data collected on 22 July 2024. Click a link to see up-to-date figures.</small>
```
Comment (lorenzolewis) [inline docs/src/content/docs/environmental-impact.md:106]:
```suggestion
<small>Data collected on 22 July 2024. Click a respective framework's link to see up-to-date figures.</small>
```
 Makes it a little more clear which link is being referred to. But may already be clear enough in this context, up to you!

**Review by HiDeoo — APPROVED** [review]
Great update, love the new rating 🌟

File: docs/src/content/docs/environmental-impact.md (hunk 1 above)
Comment (delucis) [reply docs/src/content/docs/environmental-impact.md:106]: Yeah, I suspect it's clear enough already. Although now you have me second guessing whether "click" is the right action at all if someone is on a touch device etc. 😅

Going to merge without this so I don't need to get into that, but thanks nonetheless!

**Pull request merged**